src/dagmc/DagMC.cpp

```diff
@@ -0,0 +1,774 @@
+                                      const RayHistory* history) {
+  ErrorCode rval = GQT->test_volume_boundary(volume, surface, xyz, uvw, result,
+                                             history);
+  return rval;
+}
+
+// use spherical area test to determine inside/outside of a polyhedron.
+ErrorCode DagMC::point_in_volume_slow(EntityHandle volume, const double xyz[3],
+                                      int& result) {
+  ErrorCode rval = GQT->point_in_volume_slow(volume, xyz, result);
+  return rval;
+}
+
+// detemine distance to nearest surface
+ErrorCode DagMC::closest_to_location(EntityHandle volume,
+                                     const double coords[3], double& result,
+                                     EntityHandle* surface) {
+  ErrorCode rval = GQT->closest_to_location(volume, coords, result, surface);
+  return rval;
+}
+
+// calculate volume of polyhedron
+ErrorCode DagMC::measure_volume(EntityHandle volume, double& result) {
+  ErrorCode rval = GQT->measure_volume(volume, result);
+  return rval;
+}
+
+// sum area of elements in surface
+ErrorCode DagMC::measure_area(EntityHandle surface, double& result) {
+  ErrorCode rval = GQT->measure_area(surface, result);
+  return rval;
+}
+
+// get sense of surface(s) wrt volume
+ErrorCode DagMC::surface_sense(EntityHandle volume, int num_surfaces,
+                               const EntityHandle* surfaces, int* senses_out) {
+  ErrorCode rval = GTT->get_surface_senses(volume, num_surfaces, surfaces,
+                                           senses_out);
+  return rval;
+}
+
+// get sense of surface(s) wrt volume
+ErrorCode DagMC::surface_sense(EntityHandle volume, EntityHandle surface,
+                               int& sense_out) {
+  ErrorCode rval = GTT->get_sense(surface, volume, sense_out);
+  return rval;
+}
+
+ErrorCode DagMC::get_angle(EntityHandle surf, const double in_pt[3],
+                           double angle[3],
+                           const RayHistory* history) {
+  ErrorCode rval = GQT->get_normal(surf, in_pt, angle, history);
+  return rval;
+}
+
+ErrorCode DagMC::next_vol(EntityHandle surface, EntityHandle old_volume,
+                          EntityHandle& new_volume) {
+  ErrorCode rval = GTT->next_vol(surface, old_volume, new_volume);
+  return rval;
+}
+
+/* SECTION III */
+
+EntityHandle DagMC::entity_by_id(int dimension, int id) {
+  return GTT->entity_by_id(dimension, id);
+}
+
+int DagMC::id_by_index(int dimension, int index) {
+  EntityHandle h = entity_by_index(dimension, index);
+  if (!h)
+    return 0;
+
+  int result = 0;
+  MBI->tag_get_data(GTT->get_gid_tag(), &h, 1, &result);
+  return result;
+}
+
+int DagMC::get_entity_id(EntityHandle this_ent) {
+  return GTT->global_id(this_ent);
+}
+
+ErrorCode DagMC::build_indices(Range& surfs, Range& vols) {
+  ErrorCode rval = MB_SUCCESS;
+
+  // surf/vol offsets are just first handles
+  setOffset = std::min(*surfs.begin(), *vols.begin());
```

Comment:
Error checking: Might be prudent to do a length check on `surfs` and `vols` here before getting these offsets.